src/frontends/src/query_frontend.hpp

```diff
@@ -287,6 +289,18 @@ class RequestHandler {
             request->content.string(), name, versioned_models, policy,
             latency_slo_micros, input_type);
         prediction.then([response, app_metrics](boost::future<Response> f) {
+          if (f.has_exception()) {
+            try {
+              boost::rethrow_exception(f.get_exception_ptr());
```

Comment:
Yes. If the boost future encounters a fatal exception, it will store it in an exception pointer tied to the future. `rethrow_exception` allows us to obtain the original exception that was thrown and log / handle it appropriately. This flow is a documented way of performing exception propagation when using boost futures / threads.